queries/__init__.py

```diff
@@ -253,6 +253,546 @@ def convert(m):
+        "þf": "a",
+        "þgf": "a",
+        "ef": "a",
+    },
+    "kvk": {
+        "nf": "a",
+        "þf": "u",
+        "þgf": "u",
+        "ef": "u",
+    },
+    "hk": {
+        "nf": "a",
+        "þf": "a",
+        "þgf": "a",
+        "ef": "a",
+    },
+}
+
+_TENS_NEUT_TO_ORDINAL = {
+    "tuttugu": "tuttug",
+    "þrjátíu": "þrítug",
+    "fjörutíu": "fertug",
+    "fimmtíu": "fimmtug",
+    "sextíu": "sextug",
+    "sjötíu": "sjötug",
+    "áttatíu": "átttug",
+    "níutíu": "nítug",
+}
+
+_LARGE_ORDINAL_SUFFIX = {
+    "kk": {
+        "nf": "asti",
+        "þf": "asta",
+        "þgf": "asta",
+        "ef": "asta",
+    },
+    "kvk": {
+        "nf": "asta",
+        "þf": "ustu",
+        "þgf": "ustu",
+        "ef": "ustu",
+    },
+    "hk": {
+        "nf": "asta",
+        "þf": "asta",
+        "þgf": "asta",
+        "ef": "asta",
+    },
+}
+
+
+def _num_to_ordinal(
+    word: str, decl: str = "nf", gender: str = "kk", number: str = "et"
+) -> str:
+    """
+    Helper function. Changes one part of a number (in written form) to ordinal form
+    in correct declension, gender and number.
+    Example:
+        "hundruð" -> "hundraðasti" (default args)
+        "tvö" -> "aðrar" (þf, kvk, ft)
+    """
+    if word == "núll":
+        word = "núllt" + _SUB_20_ORDINAL_SUFFIX[gender][decl]
+
+    elif word == "tvö":
+        word = _ANNAR_TABLE[number][gender][decl]
+
+    elif word in _SUB_20_NEUT_TO_ORDINAL:
+        word = _SUB_20_NEUT_TO_ORDINAL.get(word, word)
+        if number == "ft":
+            word += "u"
+        else:
+            word += _SUB_20_ORDINAL_SUFFIX[gender][decl]
+
+    elif word in _TENS_NEUT_TO_ORDINAL:
+        word = _TENS_NEUT_TO_ORDINAL.get(word, word)
+        if number == "ft":
+            word += "ustu"
+        else:
+            word += _LARGE_ORDINAL_SUFFIX[gender][decl]
+
+    elif word.startswith("hundr"):
+        if number == "ft" or (gender == "kvk" and decl != "nf"):
+            word = "hundruðustu"
+        else:
+            word = "hundrað" + _LARGE_ORDINAL_SUFFIX[gender][decl]
+
+    elif word == "þúsund":
+        if number == "ft" or (gender == "kvk" and decl != "nf"):
+            word = "þúsundustu"
+        else:
+            word = "þúsund" + _LARGE_ORDINAL_SUFFIX[gender][decl]
+
+    elif "jón" in word:
+        if number == "ft":
+            word = re.sub(r"(\S*jón)\S*", r"\1ustu", word)
+        else:
+            word = re.sub(
+                r"(\S*jón)\S*", r"\1" + _LARGE_ORDINAL_SUFFIX[gender][decl], word
+            )
+
+    elif "jarð" in word:
+        if number == "ft" or (gender == "kvk" and decl != "nf"):
+            word = re.sub(r"(\S*)jarð\S*", r"\1jörðustu", word)
+        else:
+            word = re.sub(
+                r"(\S*jarð)\S*", r"\1" + _LARGE_ORDINAL_SUFFIX[gender][decl], word
+            )
+
+    return word
+
+
+def neutral_text_to_ordinal(
+    s: str, decl: str = "nf", gender: str = "kk", number: str = "et"
```

Comment:
Suggest decl -> case

Reply:
Multiple typed assignments are kind of illegible, IMO. I'd use one line per typed var assignment.

Reply:
Also, IMO typing annotation for a string assignment is excessive. 'x = "blabla"'  is quite clear, no typing annotation needed.

Reply:
Fixed